src/views/List.jsx

```diff
@@ -1,14 +1,44 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [searchString, setSearchString] = useState('');
+
+	const handleChange = (e) => {
+		setSearchString(e.target.value);
+	};
+
+	const handleClick = (e) => {
+		e.preventDefault();
+		setSearchString('');
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+			<form>
+				<label htmlFor="searchString">
+					Search:
+					<input
+						type="text"
+						id="searchString"
+						name="searchString"
+						value={searchString}
+						onChange={handleChange}
+					/>
+				</label>
+				{searchString ? <button onClick={handleClick}>x</button> : ''}
+			</form>
+
 			<ul>
 				{data && data.length > 0 ? (
-					data.map((item, id) => <ListItem key={id} name={item.name} />)
+					data
+						.filter((d) =>
+							d.name.toLowerCase().includes(searchString.toLowerCase()),
```

Comment:
🔍 FYI, this line of code can cause errors that crash the app when an item's `name` isn't defined. We can safeguard against this with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining):

```suggestion
							d.name?.toLowerCase().includes(searchString.toLowerCase()),
```

By writing `name?.`, with the question mark, we stop this line of code from trying to call `.toLowerCase()` if `name` is `undefined`. This can sometimes happen when our list ends up in a strange state, like this:

![image](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/13525251/3d356d74-573d-42b9-ac10-0a23df0b6014)

... and _this_ can happen sometimes because the way the list is set up now differs from how it was first set up. H/t to @rmorabia for pinpointing this, and for the screenshot.